src/utils/positionUtils.ts

```diff
@@ -585,3 +585,64 @@ export const calcExecutionPriceWithImpact = ({
   if (side === PositionSide.short) numerator = numerator + (size < 0n ? -offset : offset)
   return size !== 0n ? Big6Math.abs(Big6Math.div(numerator, size)) : 0n
 }
+
+export async function waitForSettlement({
+  publicClient,
+  txHash,
+  timeoutMs = 30000,
+  onSettlement,
+}: {
+  publicClient: PublicClient
+  txHash: Hash
+  timeoutMs: number
+  onSettlement?: (txReceipt?: TransactionReceipt) => void
+}): Promise<TransactionReceipt> {
+  return new Promise(async (resolve, reject) => {
+    let timeoutId: NodeJS.Timeout
+    let unwatch: (() => void) | undefined
+
+    const cleanup = () => {
+      if (timeoutId) clearTimeout(timeoutId)
+      if (unwatch) unwatch()
+    }
+
+    try {
+      // Wait for transaction receipt
+      const receipt = await publicClient.waitForTransactionReceipt({ hash: txHash })
+
+      const logs = parseEventLogs({ logs: receipt.logs, abi: OracleAbi, eventName: 'OracleProviderVersionRequested' })
+      const oracleVersionRequestedEvent = logs[0]
+      const { version } = oracleVersionRequestedEvent.args
+
+      unwatch = publicClient.watchEvent({
+        address: oracleVersionRequestedEvent.address,
+        // event: parseAbiItem(OracleAbi, 'OracleVersionFulfilled'),
+        event: parseAbiItem(
```

Comment:
I think we can use [`watchContractEvent`](https://viem.sh/docs/contract/watchContractEvent.html) and pass in the `OracleAbi`